labrad/wrappers.py

```diff
@@ -389,18 +389,21 @@ def _send(self, *args, **kw):
 
 @inlineCallbacks
 def getConnection(host=C.MANAGER_HOST, port=None, name="Python Client",
-                  password=None, tls_mode=C.MANAGER_TLS):
+                  password=None, tls_mode=C.MANAGER_TLS, username=None,
+                  headless=False):
     """Connect to LabRAD and return a deferred that fires the protocol object."""
     p = yield protocol.connect(host, port, tls_mode)
-    yield p.authenticate(password)
+    yield p.authenticate(password, username, headless)
     yield p.loginClient(name)
     returnValue(p)
 
 @inlineCallbacks
 def connectAsync(host=C.MANAGER_HOST, port=None, name="Python Client",
-                 password=None, tls_mode=C.MANAGER_TLS):
+                 password=None, tls_mode=C.MANAGER_TLS, username=None,
+                 headless=False):
     """Connect to LabRAD and return a deferred that fires the client object."""
-    p = yield getConnection(host, port, name, password, tls_mode=tls_mode)
+    p = yield getConnection(host, port, name, password, tls_mode=tls_mode,
+                            username=username, headless=headless)
```

Comment:
This would be nice, because then we can have a "configuration factory" that
could read a config file that looked like .ssh/config and create the
connect options object for a given hostname.



On Mon, Aug 1, 2016 at 1:27 PM, Matthew Neeley <notifications@github.com>
wrote:

> In labrad/wrappers.py
> <https://github.com/labrad/pylabrad/pull/275#discussion_r73046733>:
>
> >      """Connect to LabRAD and return a deferred that fires the client object."""
> > -    p = yield getConnection(host, port, name, password, tls_mode=tls_mode)
> > +    p = yield getConnection(host, port, name, password, tls_mode=tls_mode,
> > +                            username=username, headless=headless)
>
> I agree. I was thinking about collecting these all into some sort of
> "ConnectOptions" class that would get passed around. Only the top-level
> "labrad.connect" function would accept separate keyword args. Maybe it's
> worth making that change here.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/labrad/pylabrad/pull/275/files/183bd29ea42466589ecd2bf812f1ed9c3bc57da8#r73046733>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABnKHNUK-eTnrb3M6dUP5JqtClfTBR_tks5qblapgaJpZM4JZ6cK>
> .
>